libnavigation-core/src/main/java/com/mapbox/navigation/core/CurrentIndicesSnapshotProvider.kt

```diff
@@ -0,0 +1,69 @@
+package com.mapbox.navigation.core
+
+import com.mapbox.navigation.base.internal.CurrentIndicesSnapshot
+import com.mapbox.navigation.base.trip.model.RouteProgress
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+
+/**
+ * Gets updates from [RouteProgress], converts them into [CurrentIndicesSnapshot]
+ * and provides said info.
+ * Supports freezing and unfreezing of updates, see [freezeAndGet] and [unfreeze] for details.
+ */
+internal class CurrentIndicesSnapshotProvider :
```

Comment:
just curious, why you don't used coroutines mutex? (as we already widely use coroutines 🤔 )

Reply:
https://github.com/mapbox/mapbox-navigation-android/issues/6182